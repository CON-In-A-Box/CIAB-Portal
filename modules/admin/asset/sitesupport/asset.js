/* jshint browser: true */
/* jshint -W097 */
/* globals  basicBackendRequest, apiRequest, showSpinner, hideSpinner */
/* exported  uploadAsset, startAssetUpload, populateAssetConfiguration,
             setAssetField */

'use strict';

function uploadAsset(id, file) {
  var data = file[0];
  var formData = new FormData();
  formData.append('AssetID', id);
  formData.append('AssetData', data);
  basicBackendRequest('POST', 'admin/asset', formData,
    function() {
      location.reload();
    });
}

function startAssetUpload(asset) {
  var upload = document.getElementById('assetToUpload.' + asset);
  upload.click();
}

function setAssetField(field) {
  var value = document.getElementById('config_' + field).value;
  showSpinner();
  apiRequest('PUT','admin/configuration',
    'Field=' + field + '&Value=' + value)
    .then(function() {
      hideSpinner();
    })
    .catch(function(response) {
      hideSpinner();
      if (response instanceof Error) { throw response; }
    });
}

function createTableCell() {
  var cell = document.createElement('DIV');
  cell.classList.add('UI-table-cell');
  cell.classList.add('UI-padding');
  return cell;
}

function populateAssetConfiguration() {
  basicBackendRequest('GET', 'admin/asset', 'configuration=1',
    function(request) {
      var base = document.getElementById('config_table');
      base.innerHTML = '';
      var output = JSON.parse(request.responseText);
      for (var property in output) {
        var line = document.createElement('DIV');
        line.classList.add('UI-table-row');
        line.classList.add('UI-padding');
        var value = createTableCell();
        value.innerHTML = property;
        line.appendChild(value);
        value = createTableCell();
        value.innerHTML = output[property].description;
        line.appendChild(value);

        if (output[property].source == 'ENV' || output[property].readonly) {
          value = createTableCell();
          value.innerHTML = output[property].value;
          line.appendChild(value);

          if (!output[property].readonly) {
            value = createTableCell();
            value.innerHTML = '&lt;ENVIRONMENT&gt;';
            line.appendChild(value);
          } else {
            value = createTableCell();
            value.innerHTML = '';
            line.appendChild(value);
          }
        } else {
          value = createTableCell();
          var edit = document.createElement('INPUT');
          edit.setAttribute('type', 'text');
          edit.id = 'config_' + property;
          edit.value = output[property].value;
          edit.classList.add('UI-input');
          value.appendChild(edit);
          line.appendChild(value);

          value = createTableCell();
          var button = document.createElement('BUTTON');
          button.classList.add('UI-eventbutton');
          button.setAttribute('onclick', 'setAssetField("' + property + '");');
          button.innerHTML = 'Set';
          value.appendChild(button);
          line.appendChild(value);
        }

        base.appendChild(line);

        if (property == 'ASSET_TYPES') {
          var upload = document.getElementById('assetToUpload.org-icon');
          upload.setAttribute('accept', output[property].value);
        }
      }
    });

}
