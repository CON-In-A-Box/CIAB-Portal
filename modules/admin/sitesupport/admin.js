/* jshint browser: true */
/* jshint -W097 */
/* globals apiRequest, confirmbox, basicBackendRequest, showSpinner,
           hideSpinner, alertbox */
/* exported downloadLog, setField, addField, removeAdmin, updateMemberships,
            doSUDO, rebuildSCSS, setPassword */

function downloadLog() {
  window.location = 'index.php?Function=admin&downloadLog=db';
}

function basicAdminRequest(parameter, finish) {
  basicBackendRequest('POST', 'admin', parameter, finish);
}

function setField(field) {
  var value = document.getElementById('config_' + field).value;
  value = btoa(value);
  basicAdminRequest('&setField=' + field + '&value=' + value, function() {
    location.reload();
  });
}

function addField() {
  var field = document.getElementById('new_Field').value;
  var value = document.getElementById('new_Value').value;
  value = btoa(value);
  basicAdminRequest('&newField=' + field + '&value=' + value, function() {
    location.reload();
  });
}

function removeAdmin() {
  basicAdminRequest('&removeAdmin=true', function() {
    location.reload();
  });
}

function updateMemberships() {
  confirmbox('Update Memberships?', 'This operations will run in the ' +
    'background and can take several minutes to complete.' +
    '<br><em>Run Command?</em>').then(function() {
    basicAdminRequest('&updateMemberships=true', function() {});
  });
}

function doSUDO() {
  var value = document.getElementById('sudo_id').value;
  apiRequest('POST', 'admin/SUDO/' + value, '');
}

function rebuildSCSS() {
  basicAdminRequest('&rebuildSCSS=true', function() {
    location.reload();
  });
}

function setPassword() {
  var newPassword = document.getElementById('tmp_passwd').value;
  var account = document.getElementById('tmp_login').value;

  showSpinner();
  apiRequest('PUT', 'member/' + account + '/password' ,
    'NewPassword=' + newPassword)
    .then(function() {
      hideSpinner();
      alertbox('Password Updated');
    })
    .catch(function(response) {
      hideSpinner();
      alertbox('Password Reset Failed: ' + response.responseText);
    });
}
