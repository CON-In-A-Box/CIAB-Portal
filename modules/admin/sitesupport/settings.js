/*
 * Javacript for the Site Configuration page
 */

/* jshint browser: true */
/* jshint -W097 */
/* globals settingsTable */

function load() {
  new settingsTable({api: 'admin/configuration'}).createElement();
}

if (window.addEventListener) {
  window.addEventListener('load', load);
} else {
  window.attachEvent('onload', load);
}
