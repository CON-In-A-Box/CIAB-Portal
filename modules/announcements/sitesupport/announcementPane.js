/*
 * Javacript for the Announcements Main Panel
 */

/* jshint browser: true */
/* jshint -W097 */
/* globals apiRequest */

var announcementPane = (function(options) {
  'use strict';

  var settings = Object.assign(
    {
    }, options);

  return {
    options: function(opts) {
      settings = Object.assign(settings, opts);
    },

    displayAnnouncements: function(frame) {
      var target = document.getElementById('announcement_pane');
      if (frame !== null) {
        target.appendChild(frame[0]);
      }
    },

    addAnnouncement: function(table, data2) {
      var line = document.createElement('DIV');
      line.classList.add('UI-table-row');
      line.classList.add('UI-white');
      var f = document.createElement('DIV');
      f.classList.add('UI-table-cell');
      f.classList.add('UI-announcement');
      f.id = 'announcement-' + data2.id + '-note';

      var a = document.createElement('DIV');
      a.appendChild(document.createTextNode(data2.text));
      var s = document.createElement('SPAN');
      s.classList.add('UI-announcement-source');
      s.appendChild(document.createTextNode('  -  ' + data2.departmentId.name));
      a.appendChild(s);
      s.appendChild(document.createTextNode(' (' + data2.postedOn + ')'));
      a.appendChild(s);
      f.appendChild(a);
      line.appendChild(f);
      table.appendChild(line);
      return 0;
    },

    emptyAnnouncementBlock: function() {
      var block = document.createElement('DIV');
      block.id = 'announcement-block';
      block.classList.add('UI-container');
      block.classList.add('event-color-primary');
      var title = document.createElement('div');
      title.classList.add('event-color-primary');
      title.classList.add('UI-center');
      var txt = document.createElement('B');
      txt.classList.add('UI-bold');
      txt.style.fontSize = '125%';
      txt.appendChild(document.createTextNode('Announcements'));
      title.appendChild(txt);
      block.appendChild(title);
      var contents = document.createElement('DIV');
      contents.classList.add('UI-container');
      block.appendChild(contents);
      var table = document.createElement('DIV');
      table.classList.add('UI-table');
      table.classList.add('UI-table-padded');
      contents.appendChild(table);
      return [block, table];
    },

    buildAnnouncementBlock: function(frame, result, data) {
      var rc = frame[0];
      var table = rc[1];
      return announcementPane.addAnnouncement(table, data);
    },

    load: function() {
      var frame = [ announcementPane.emptyAnnouncementBlock() ];
      apiRequest('GET',
        'member/current/announcements',
        'maxResults=all&include=departmentId')
        .then(function(response) {
          var target = document.getElementById('announcement_pane');
          var result = JSON.parse(response.responseText);
          if (result.data.length > 0) {
            result.data.forEach(function(data) {
              announcementPane.buildAnnouncementBlock(frame, result, data);
            });
            var line = document.createElement('DIV');
            line.classList.add('event-color-primary');
            line.appendChild(document.createTextNode('\u00A0'));
            frame[0][0].appendChild(line);
            target.classList.remove('UI-hide');
          } else {
            target.classList.add('UI-hide');
          }
        });
      announcementPane.displayAnnouncements(frame[0]);
    }
  };
})();

if (window.addEventListener) {
  window.addEventListener('load', announcementPane.load);
} else {
  window.attachEvent('onload', announcementPane.load);
}
