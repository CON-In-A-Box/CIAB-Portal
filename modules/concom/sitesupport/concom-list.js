/* globals Vue, showSpinner, hideSpinner, Vuetify */
import staffListComponent from './components/staff-list.js';
import sectionNavComponent from './components/staff-section-nav.js';
import staffDivisionComponent from './components/staff-division.js';
import departmentHeaderComponent from './components/department-header.js';
import departmentMemberComponent from './components/department-member.js';
import staffDepartmentComponent from './components/staff-department.js';
import staffSidebarComponent from './components/staff-sidebar.js';
import lookupuser from '../../../sitesupport/vue/lookupuser.js';
import staffDivisionVisualComponent from './components/staff-division-visual.js';

function updateLoading() {
  this.isLoading = !this.isLoading;
  this.isLoading ? showSpinner() : hideSpinner();
}

const staffStandardTheme = {
  dark: false,
  colors: {
    primary: '#620272',
    secondary: '#00e500',
    error: '#B00020',
    info: '#2196F3',
    success: '#4CAF50',
    warning: '#FB8C00'
  }
}

const vuetify = Vuetify.createVuetify({
  theme: {
    defaultTheme: 'staffStandardTheme',
    themes: {
      staffStandardTheme
    }
  },
  defaults: {
    VBtn: {
      color: 'primary'
    },
    VSelect: {
      variant: 'outlined'
    },
    VListItem: {
      'color': 'white'
    }
  }
});

const staffApp = Vue.createApp({
  setup() {
    const isLoading = Vue.ref(false);
    Vue.provide('isLoading', {
      isLoading,
      updateLoading
    });

    return {
      isLoading
    }
  },
  methods: {
    updateLoading
  }
}).use(vuetify);

staffApp.component('staff-division', staffDivisionComponent);
staffApp.component('staff-section-nav', sectionNavComponent);
staffApp.component('staff-list', staffListComponent);
staffApp.component('department-header', departmentHeaderComponent);
staffApp.component('department-member', departmentMemberComponent);
staffApp.component('staff-department', staffDepartmentComponent);
staffApp.component('staff-sidebar', staffSidebarComponent);
staffApp.component('lookup-user', lookupuser);
staffApp.component('org-donut', staffDivisionVisualComponent);

staffApp.mount('#staff-app');

export default staffApp;
