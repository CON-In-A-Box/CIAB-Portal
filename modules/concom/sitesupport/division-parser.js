const extractDepartment = (item) => {
  const hasParent = item.parent !== null;

  return {
    id: item.id,
    name: item.name,
    email: item.email,
    ...hasParent && {
      parent_name: item.parent.name
    }
  }
}

const extractDivisions = (departmentData) => {
  return departmentData.filter((item) => item.parent === null)
    .map((item) => {
      const mappedDivision = extractDepartment(item);
      return {
        ...mappedDivision,
        departments: [ mappedDivision ]
      }
    })
    .reduce((prev, current) => {
      prev[current.name] = current;

      return prev;
    }, {});
}

/**
 *
 * Data comes back in format as an array:
 *
 * {
 *   "id": "123",
 *   "parent": null OR {
 *     "id": "456",
 *     "parent": null OR nested department
 *     "name": "Parent Department",
 *     "fallback": null OR,
 *     "child_count": "2",
 *     "email": [
 *       "parent-dept@test-con.org"
 *     ],
 *     "type": "department"
 *   },
 *   "name": "Department A",
 *   "fallback": null OR,
 *   "child_count": "9",
 *   "email": [
 *     "department-a@test-con.org"
 *   ]
 * }
 *
 *
 * @param {*} departmentData
 * @returns array of divisions
 */
export const extractDivisionHierarchy = (departmentData) => {
  const divisionHierarchy = extractDivisions(departmentData);

  // Extract department data and apply to appropriate division based on parent name.
  departmentData.filter((item) => item.parent !== null)
    .map((item) => {
      return extractDepartment(item);
    })
    .forEach((item) => {
      const parentDivision = divisionHierarchy[item.parent_name];
      if (parentDivision !== undefined) {
        parentDivision.departments.push(item);
      }
    });

  return Object.keys(divisionHierarchy).map((item) => {
    return divisionHierarchy[item]
  });
}
