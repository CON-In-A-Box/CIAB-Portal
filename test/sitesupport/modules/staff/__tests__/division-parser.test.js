/* globals describe, it, expect */
import { extractDivisionHierarchy, createDonutData } from '../../../../../modules/concom/sitesupport/division-parser';
import departmentList from './department_list.json';
import departmentListCircular from './department_list_circular_fallback.json';

const verifyDonutData = (data, fallbackDepts) => {
  data.forEach((item, idx) => {
    // Division at top level node data.
    expect(item.nodeData.colorIndex).toEqual(idx);
    expect(item.nodeData.dsize).toEqual(1);
    expect(item.nodeData.strokeWidth).toEqual(3);
    expect(item.nodeData).toHaveProperty('label');
    expect(item.nodeData.link.includes(item.nodeData.label.replaceAll(' ', '_'))).toEqual(true);

    const arrowNode = item.subData[0];
    // First level of "subData" always has one element. An arrow and then departments.
    expect(arrowNode.nodeData.colorIndex).toEqual(idx);
    expect(arrowNode.nodeData.dsize).toEqual(1);
    expect(arrowNode.nodeData.noRotate).toEqual(1);
    expect(arrowNode.nodeData.pieWidth).toEqual(10);
    expect(arrowNode.nodeData.strokeWidth).toEqual(2);

    if (fallbackDepts.includes(item.nodeData.label)) {
      expect(arrowNode.nodeData.label).toEqual('↓');
    } else {
      expect(arrowNode.nodeData.label).toEqual('⌾');
    }

    // Departments
    arrowNode.subData.forEach((dept) => {
      expect(dept.nodeData.colorIndex).toEqual(idx);
      expect(dept.nodeData.dsize).toBeCloseTo((1 / arrowNode.subData.length), 5);
      expect(dept.nodeData.strokeWidth).toEqual(1);
      expect(dept.nodeData).toHaveProperty('label');
      expect(dept.nodeData.link.includes(dept.nodeData.label.replaceAll(' ', '_'))).toEqual(true);
    });
  });
}

describe('Staff Division Parser', () => {
  it('can parse the division hierarchy', () => {
    const result = extractDivisionHierarchy(departmentList.data);
    expect(result).toMatchSnapshot();
  });

  it('can parse donut data from the division data', () => {
    const divisions = extractDivisionHierarchy(departmentList.data);
    const result = createDonutData(divisions);

    expect(divisions.length === result.length).toEqual(true);

    const fallbackDeptNames = ['Activities', 'Administration', 'External Relations and Communications',
      'Facilities', 'Productions', 'Systems'];
    verifyDonutData(result, fallbackDeptNames);
  });

  it('can parse donut data when no fallbacks are present', () => {
    const departmentDataNoFallbacks = departmentList.data.map((item) => {
      return {
        ...item,
        fallback: null
      }
    });

    const divisions = extractDivisionHierarchy(departmentDataNoFallbacks);
    const result = createDonutData(divisions);
    result.forEach((item) => {
      const arrowNode = item.subData[0];
      expect(arrowNode.nodeData.label).toEqual('⌾');
    });
  });

  it('can parse donut data when fallbacks create a circular reference', () => {
    const divisions = extractDivisionHierarchy(departmentListCircular.data);
    const result = createDonutData(divisions);

    expect(divisions.length === result.length).toEqual(true);

    const fallbackDeptNames = ['Activities', 'Administration', 'External Relations and Communications', 'Facilities', 'Hospitality', 'Productions',
      'Systems'];
    verifyDonutData(result, fallbackDeptNames);
  });
});
