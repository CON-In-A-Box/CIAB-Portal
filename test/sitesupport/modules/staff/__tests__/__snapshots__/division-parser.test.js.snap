// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Staff Division Parser can parse the division hierarchy 1`] = `
[
  {
    "departments": [
      {
        "email": [
          "activities@test-con.org",
        ],
        "id": "1",
        "name": "Activities",
      },
      {
        "email": [
          "bookswap@test-con.org",
        ],
        "id": "105",
        "name": "Book Swap",
        "parent_name": "Activities",
      },
      {
        "email": [
          "sandbox@test-con.org",
        ],
        "id": "110",
        "name": "Connies Quantum Sandbox",
        "parent_name": "Activities",
      },
      {
        "email": [
          "exhibits@test-con.org",
        ],
        "id": "117",
        "name": "Exhibits",
        "parent_name": "Activities",
      },
      {
        "email": [
          "gaming@test-con.org",
        ],
        "id": "121",
        "name": "Gaming",
        "parent_name": "Activities",
      },
      {
        "email": [
          "pros@test-con.org",
        ],
        "id": "126",
        "name": "Invited Participants",
        "parent_name": "Activities",
      },
      {
        "email": [
          "programming@test-con.org",
        ],
        "id": "140",
        "name": "Programming",
        "parent_name": "Activities",
      },
      {
        "email": [
          "teens@test-con.org",
        ],
        "id": "149",
        "name": "Teen Room",
        "parent_name": "Activities",
      },
      {
        "email": [],
        "id": "160",
        "name": "Spoken Word",
        "parent_name": "Activities",
      },
    ],
    "email": [
      "activities@test-con.org",
    ],
    "id": "1",
    "name": "Activities",
  },
  {
    "departments": [
      {
        "email": [
          "administration@test-con.org",
        ],
        "id": "2",
        "name": "Administration",
      },
      {
        "email": [
          "archives@test-con.org",
        ],
        "id": "102",
        "name": "Archives",
        "parent_name": "Administration",
      },
      {
        "email": [
          "artshow@test-con.org",
        ],
        "id": "104",
        "name": "Art Show",
        "parent_name": "Administration",
      },
      {
        "email": [
          "guests@test-con.org",
        ],
        "id": "123",
        "name": "Guests of Honor",
        "parent_name": "Administration",
      },
      {
        "email": [
          "merchandise@test-con.org",
        ],
        "id": "133",
        "name": "Merchandise",
        "parent_name": "Administration",
      },
      {
        "email": [
          "partners@test-con.org",
        ],
        "id": "136",
        "name": "Partner Management",
        "parent_name": "Administration",
      },
      {
        "email": [
          "registration@test-con.org",
        ],
        "id": "142",
        "name": "Registration",
        "parent_name": "Administration",
      },
      {
        "email": [
          "volunteers@test-con.org",
        ],
        "id": "156",
        "name": "Volunteers",
        "parent_name": "Administration",
      },
    ],
    "email": [
      "administration@test-con.org",
    ],
    "id": "2",
    "name": "Administration",
  },
  {
    "departments": [
      {
        "email": [
          "erac@test-con.org",
        ],
        "id": "3",
        "name": "External Relations and Communications",
      },
      {
        "email": [
          "asl@test-con.org",
          "accessibility@test-con.org",
        ],
        "id": "100",
        "name": "Accessibility and Inclusion",
        "parent_name": "External Relations and Communications",
      },
      {
        "email": [
          "artistalley@test-con.org",
        ],
        "id": "103",
        "name": "Artist Alley",
        "parent_name": "External Relations and Communications",
      },
      {
        "email": [
          "creativeservices@test-con.org",
        ],
        "id": "112",
        "name": "Creative Services",
        "parent_name": "External Relations and Communications",
      },
      {
        "email": [
          "dealers@test-con.org",
        ],
        "id": "114",
        "name": "Dealers Room",
        "parent_name": "External Relations and Communications",
      },
      {
        "email": [
          "photography@test-con.org",
        ],
        "id": "138",
        "name": "Photography",
        "parent_name": "External Relations and Communications",
      },
      {
        "email": [
          "publications@test-con.org",
        ],
        "id": "141",
        "name": "Publications",
        "parent_name": "External Relations and Communications",
      },
      {
        "email": [
          "socialmedia@test-con.org",
        ],
        "id": "147",
        "name": "Social Media",
        "parent_name": "External Relations and Communications",
      },
      {
        "email": [
          "webteam@test-con.org",
        ],
        "id": "157",
        "name": "Webteam",
        "parent_name": "External Relations and Communications",
      },
    ],
    "email": [
      "erac@test-con.org",
    ],
    "id": "3",
    "name": "External Relations and Communications",
  },
  {
    "departments": [
      {
        "email": [
          "facilities@test-con.org",
        ],
        "id": "4",
        "name": "Facilities",
      },
      {
        "email": [
          "allocations@test-con.org",
        ],
        "id": "101",
        "name": "Allocations",
        "parent_name": "Facilities",
      },
      {
        "email": [
          "decor@test-con.org",
        ],
        "id": "115",
        "name": "Decor",
        "parent_name": "Facilities",
      },
      {
        "email": [
          "dock@test-con.org",
        ],
        "id": "116",
        "name": "Dock",
        "parent_name": "Facilities",
      },
      {
        "email": [
          "hotel@test-con.org",
        ],
        "id": "125",
        "name": "Hotel",
        "parent_name": "Facilities",
      },
      {
        "email": [
          "logistics@test-con.org",
        ],
        "id": "128",
        "name": "Logistics",
        "parent_name": "Facilities",
      },
      {
        "email": [
          "childcare@test-con.org",
        ],
        "id": "131",
        "name": "Meeting Childcare",
        "parent_name": "Facilities",
      },
      {
        "email": [
          "nerfherders@test-con.org",
        ],
        "id": "134",
        "name": "Nerf Herders",
        "parent_name": "Facilities",
      },
      {
        "email": [
          "eventorders@test-con.org",
        ],
        "id": "143",
        "name": "Event Orders",
        "parent_name": "Facilities",
      },
      {
        "email": [
          "transportation@test-con.org",
        ],
        "id": "151",
        "name": "Transportation",
        "parent_name": "Facilities",
      },
    ],
    "email": [
      "facilities@test-con.org",
    ],
    "id": "4",
    "name": "Facilities",
  },
  {
    "departments": [
      {
        "email": [
          "hospitality@test-con.org",
        ],
        "id": "5",
        "name": "Hospitality",
      },
      {
        "email": [
          "cof2e2@test-con.org",
        ],
        "id": "109",
        "name": "CoF2E2",
        "parent_name": "Hospitality",
      },
      {
        "email": [
          "consuite@test-con.org",
        ],
        "id": "111",
        "name": "ConSuite",
        "parent_name": "Hospitality",
      },
      {
        "email": [
          "firstadvisors@test-con.org",
        ],
        "id": "119",
        "name": "First Advisors",
        "parent_name": "Hospitality",
      },
      {
        "email": [
          "operations@test-con.org",
        ],
        "id": "135",
        "name": "Operations",
        "parent_name": "Hospitality",
      },
      {
        "email": [
          "roomparties@test-con.org",
        ],
        "id": "144",
        "name": "Room Parties",
        "parent_name": "Hospitality",
      },
      {
        "email": [
          "smokersparadise@test-con.org",
        ],
        "id": "146",
        "name": "Smokers Paradise",
        "parent_name": "Hospitality",
      },
      {
        "email": [
          "spacelounge@test-con.org",
        ],
        "id": "148",
        "name": "Space Lounge",
        "parent_name": "Hospitality",
      },
      {
        "email": [
          "volden@test-con.org",
        ],
        "id": "155",
        "name": "Volunteer Den",
        "parent_name": "Hospitality",
      },
    ],
    "email": [
      "hospitality@test-con.org",
    ],
    "id": "5",
    "name": "Hospitality",
  },
  {
    "departments": [
      {
        "email": [
          "productions@test-con.org",
        ],
        "id": "6",
        "name": "Productions",
      },
      {
        "email": [
          "ceremonies@test-con.org",
        ],
        "id": "106",
        "name": "Ceremonies",
        "parent_name": "Productions",
      },
      {
        "email": [
          "cinemarex@test-con.org",
        ],
        "id": "108",
        "name": "Cinema Rex",
        "parent_name": "Productions",
      },
      {
        "email": [
          "cvgtv@test-con.org",
        ],
        "id": "113",
        "name": "CVG-TV",
        "parent_name": "Productions",
      },
      {
        "email": [
          "harmonic@test-con.org",
        ],
        "id": "124",
        "name": "Harmonic Convergence",
        "parent_name": "Productions",
      },
      {
        "email": [
          "mainstage@test-con.org",
        ],
        "id": "129",
        "name": "MainStage",
        "parent_name": "Productions",
      },
      {
        "email": [
          "masquerade@test-con.org",
        ],
        "id": "130",
        "name": "Masquerade",
        "parent_name": "Productions",
      },
      {
        "email": [
          "theaternippon@test-con.org",
        ],
        "id": "150",
        "name": "Theater Nippon",
        "parent_name": "Productions",
      },
      {
        "email": [
          "video@test-con.org",
        ],
        "id": "154",
        "name": "Video",
        "parent_name": "Productions",
      },
    ],
    "email": [
      "productions@test-con.org",
    ],
    "id": "6",
    "name": "Productions",
  },
  {
    "departments": [
      {
        "email": [
          "systems@test-con.org",
        ],
        "id": "7",
        "name": "Systems",
      },
      {
        "email": [
          "it@test-con.org",
        ],
        "id": "127",
        "name": "IT",
        "parent_name": "Systems",
      },
    ],
    "email": [
      "systems@test-con.org",
    ],
    "id": "7",
    "name": "Systems",
  },
  {
    "departments": [
      {
        "email": [
          "info@test-con.org",
        ],
        "id": "8",
        "name": "Committees",
      },
      {
        "email": [
          "futurevisioning@test-con.org",
        ],
        "id": "120",
        "name": "Future Visioning",
        "parent_name": "Committees",
      },
      {
        "email": [
          "guestsearch@test-con.org",
        ],
        "id": "122",
        "name": "Guest Search",
        "parent_name": "Committees",
      },
      {
        "email": [
          "memberadvocates@test-con.org",
        ],
        "id": "132",
        "name": "Member Advocates",
        "parent_name": "Committees",
      },
      {
        "email": [
          "partnersearch@test-con.org",
        ],
        "id": "137",
        "name": "Partner Search",
        "parent_name": "Committees",
      },
      {
        "email": [
          "youthprogramming@test-con.org",
        ],
        "id": "158",
        "name": "Youth Programming Advisory",
        "parent_name": "Committees",
      },
      {
        "email": [
          "transitionteam@test-con.org",
        ],
        "id": "159",
        "name": "Transition Team",
        "parent_name": "Committees",
      },
    ],
    "email": [
      "info@test-con.org",
    ],
    "id": "8",
    "name": "Committees",
  },
  {
    "departments": [
      {
        "email": [
          "directors@test-con.org",
        ],
        "id": "9",
        "name": "Corporate Staff",
      },
      {
        "email": [
          "president@test-con.org",
        ],
        "id": "139",
        "name": "President",
        "parent_name": "Corporate Staff",
      },
      {
        "email": [
          "secretary@test-con.org",
        ],
        "id": "145",
        "name": "Secretary",
        "parent_name": "Corporate Staff",
      },
      {
        "email": [
          "treasurer@test-con.org",
        ],
        "id": "152",
        "name": "Treasurer",
        "parent_name": "Corporate Staff",
      },
      {
        "email": [
          "vicepresident@test-con.org",
        ],
        "id": "153",
        "name": "Vice-President",
        "parent_name": "Corporate Staff",
      },
    ],
    "email": [
      "directors@test-con.org",
    ],
    "id": "9",
    "name": "Corporate Staff",
  },
  {
    "departments": [
      {
        "email": [
          "cfo@test-con.org",
        ],
        "id": "10",
        "name": "CFO Staff",
      },
      {
        "email": [
          "finance@test-con.org",
        ],
        "id": "118",
        "name": "Finance",
        "parent_name": "CFO Staff",
      },
    ],
    "email": [
      "cfo@test-con.org",
    ],
    "id": "10",
    "name": "CFO Staff",
  },
]
`;
